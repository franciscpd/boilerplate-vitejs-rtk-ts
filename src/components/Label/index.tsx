import clx from 'classnames'

interface Props {
  children: String
  error?: boolean
  htmlFor?: string
  alt?: boolean
  className?: string
}

function Label({
  children,
  error = false,
  htmlFor,
  alt = false,
  className,
  ...props
}: Props) {
  return (
    <label className="label" htmlFor={htmlFor} {...props}>
      <span
        className={clx(
          {
            'label-text': !alt,
            'label-text-alt': alt,
            'text-error': error,
            'text-primary': !error && !className?.includes('text-'),
          },
          className
        )}
      >
        {children}
      </span>
    </label>
  )
}

export default Label
