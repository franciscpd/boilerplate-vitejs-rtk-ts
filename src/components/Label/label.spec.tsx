import React from 'react'
import { describe, test, expect } from 'vitest'

import { render } from '../../tests/utils'

import Label from '.'

describe('Label', () => {
  test('Should render the label', () => {
    const { getByText, container } = render(<Label>Email</Label>)

    expect(getByText('Email')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  test('Should render color primary correct', () => {
    const { container } = render(<Label>Email</Label>)

    expect(container.querySelector('span')).toHaveClass('text-primary')
  })

  test('Should render color error correct', () => {
    const { container } = render(<Label error>Email</Label>)

    expect(container.querySelector('span')).toHaveClass('text-error')
  })

  test('Should render custom color correct', () => {
    const { container } = render(<Label className="text-info">Email</Label>)

    expect(container.querySelector('span')).toHaveClass('text-info')
  })
})
