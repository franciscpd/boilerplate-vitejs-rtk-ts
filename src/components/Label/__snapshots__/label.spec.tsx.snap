// Vitest Snapshot v1

exports[`Label > Should render the label 1`] = `
<label
  class="label"
>
  <span
    class="label-text text-primary"
  >
    Email
  </span>
</label>
`;
